include/triton/core/tritoncache.h

```diff
@@ -0,0 +1,294 @@
+/// \return a TRITONSERVER_Error indicating success or failure.
+
+// TODO: Not currently using this API. See TritonCache Lookup/Insert
+// for explanation. Should clean up / clarify the expected ownerships here.
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryNew(
+    TRITONCACHE_CacheEntry** entry);
+
+/// Delete a cache entry object.
+///
+/// \param entry The cache entry object to delete.
+/// \return a TRITONSERVER_Error indicating success or failure.
+
+// TODO: Not currently using this API. See TritonCache Lookup/Insert
+// for explanation. Should clean up / clarify the expected ownerships here.
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryDelete(
+    TRITONCACHE_CacheEntry* entry);
+
+///
+/// CacheEntry Field Management
+///
+
+// TODO: Proper API descriptions, lifetimes, copies, etc.
+
+// Get number of items in entry
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemCount(
+    TRITONCACHE_CacheEntry* entry, size_t* count);
+
+// Adds item to entry
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryAddItem(
+    TRITONCACHE_CacheEntry* entry, TRITONCACHE_CacheEntryItem* item);
+
+// Gets item at index from entry where 0 <= index < count and
+// 'count' is the value returned by TRITONCACHE_CacheEntryItemCount
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryGetItem(
+    TRITONCACHE_CacheEntry* entry, size_t index,
+    TRITONCACHE_CacheEntryItem** item);
+
+
+///
+/// CacheEntryItem Lifetime Management
+///
+
+/// Create a new cache entry item object. The caller takes ownership of the
+/// TRITONCACHE_CacheEntryItem object and must call
+/// TRITONCACHE_CacheEntryItemDelete to release the object.
+///
+/// \param item Returns the new cache entry item object.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemNew(
+    TRITONCACHE_CacheEntryItem** item);
+
+/// Delete a cache entry item object.
+///
+/// \param item The cache entry item object to delete.
+/// \return a TRITONSERVER_Error indicating success or failure.
+
+// TODO: Not currently using this API. See TritonCache Lookup/Insert
+// for explanation. Should clean up / clarify the expected ownerships here.
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemDelete(
+    TRITONCACHE_CacheEntryItem* item);
+
+///
+/// CacheEntryItem Field Management
+///
+
+// TODO: Proper API descriptions, lifetimes, copies, etc.
+
+// Returns number of buffers held by item
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemBufferCount(
+    TRITONCACHE_CacheEntryItem* item, size_t* count);
+
+// Adds buffer of specified memory_type/id to item. Only CPU memory supported
+// currently.
+TRITONCACHE_DECLSPEC TRITONSERVER_Error* TRITONCACHE_CacheEntryItemAddBuffer(
+    TRITONCACHE_CacheEntryItem* item, const void* base, size_t byte_size,
+    TRITONSERVER_MemoryType memory_type, int64_t memory_type_id);
```

Comment:
Use buffer attributes API so that we can stay backwards compatible, in case we decide to add new fields to the buffers: https://github.com/triton-inference-server/core/blob/e0c85317a08565bfdccc73eed490734c49887098/include/triton/core/tritonserver.h#L1428-L1441